sql/2023/08/inline-script-types.sql

```diff
@@ -0,0 +1,70 @@
+# HTTP Archive query to get counts for types of WordPress inline scripts.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/64
+CREATE TEMP FUNCTION GET_INLINE_SCRIPT_TYPES (custom_metrics STRING) RETURNS ARRAY<STRING> LANGUAGE js AS '''
+
+/**
+ * Get script types.
+ *
+ * @param {object} data
+ * @param {object} data.cms
+ * @param {object} data.cms.wordpress
+ * @param {Array<{after_script_size: number, before_script_size: number, extra_script_size: number, translations_script_size: number}>} data.cms.wordpress.scripts
+ * @return {Array} Script types.
+ */
+function getScriptTypes(data) {
+  const scriptTypes = [];
+  for (const script of data.cms.wordpress.scripts) {
+    scriptTypes.push( (new Array( script.after_script_size )).fill( 'after' ) );
+    scriptTypes.push( (new Array( script.before_script_size )).fill( 'before' ) );
+    scriptTypes.push( (new Array( script.extra_script_size )).fill( 'extra' ) );
+    scriptTypes.push( (new Array( script.translations_script_size )).fill( 'translations' ) );
```

Comment:
Since these metrics are already numbers, it does seem like this should rather be implemented to return a `STRUCT` and utilize `SUM()` rather than `COUNT()` in the query.